proxy/http/HttpConfig.cc

```diff
@@ -1603,3 +1606,137 @@ HttpConfig::parse_ports_list(char *ports_string)
   }
   return (ports_list);
 }
+
+////////////////////////////////////////////////////////////////
+//
+//  HttpConfig::parse_redirect_actions()
+//
+////////////////////////////////////////////////////////////////
+IpMap *
+HttpConfig::parse_redirect_actions(char *input_string, RedirectEnabled::Action &self_action)
+{
+  using RedirectEnabled::Action;
+  using RedirectEnabled::AddressClass;
+  using RedirectEnabled::action_map;
+  using RedirectEnabled::address_class_map;
+
+  if (nullptr == input_string) {
+    Emergency("parse_redirect_actions: The configuration value is empty.");
```

Comment:
Emergency isn't used in the code.  I would go with Error.  That is what we do with configuration parsing.

Reply:
I think we originally had an `ink_release_assert` here, and @SolidWallOfCode suggested we use `Emergency` instead. What this appeared to do was indicate a non-recoverable error. I think `Error` will not do the same, but I could be mistaken. In this case the error is not recoverable.

Reply:
@d2r If you want the process to die on a error I would use Fatal. It is used all over traffic server for logging Fatal errors and exiting.

Update: Talked to @d2r on slack that Emergency exists with UNRECOVERABLE_EXIT and traffic_manager will not restart the process.